source/http/source.go

```diff
@@ -186,7 +202,11 @@ func (hs *httpSourceHandler) CacheKey(ctx context.Context, g session.Group, inde
 		return "", "", nil, false, err
 	}
 	req = req.WithContext(ctx)
-	req.Header.Add("User-Agent", version.UserAgent())
+	req.Header.Set("User-Agent", version.UserAgent())
+	for _, field := range hs.src.Header {
```

Comment:
I don't see any restriction atm of what keys are allowed on server side.

Reply:
I can add that. However, it will probably require a capability for each supported header, right?

This has me thinking: The encoding and cache-control headers are now added after the user-provided headers using `Header.Set` instead of `Header.Add`, so they will effectively overwrite anything provided by the user and no longer be subject to meddling. Given that, should we still limit user-provided headers?

Reply:
I think we should still limit. The current ones can have one cap, but if support for a new header(combination) is added then a new cap would be needed as well.